packages/ui/src/Slider.tsx

```diff
@@ -0,0 +1,343 @@
+   * We also send an event that triggered the change
+   */
+  const updateValues = useCallback(
+    ([changedFirstValue, changedSecondValue]: [number | null, number | null], e: ChangeEvent<HTMLInputElement> | MouseEvent) => {
+      if (isRangeGuard(value, onChange)) {
+        onChange([changedFirstValue ?? firstValue, changedSecondValue ?? secondValue], e)
+      }
+      if (changedFirstValue !== null && isSingleValueGuard(value, onChange)) {
+        onChange(changedFirstValue, e)
+      }
+    },
+    [firstValue, secondValue, onChange, value],
+  )
+
+  /**
+   * The following code makes sure that the relevant thumb is moved to the position once
+   * user clicks on a track.
+   *
+   * 1. We always move the first thumb in a non-range mode.
+   * 2. Otherwise, we move the thumb that is closed to a click event.
+   * 3. We focus the thumb after it is clicked.
+   */
+  const clickOnTrackHandler = useCallback(
+    (e: MouseEvent) => {
+      const { offsetX, target } = e
+      if (disabled) {
+        return false
+      }
+      if (!wrapperRef.current || wrapperRef.current !== target) {
+        return
+      }
+      const clickPercentage = offsetX / wrapperRef.current.offsetWidth
+      const resultStepSpot: number = Math.round((clickPercentage * max) / step) * step
+      if (!isRange || Math.abs(firstValue - resultStepSpot) < Math.abs(secondValue - resultStepSpot)) {
+        updateValues([resultStepSpot, null], e)
+        firstRangeInputRef.current?.focus()
+      } else {
+        // otherwise, let's move the right one
+        updateValues([null, resultStepSpot], e)
+        secondRangeInputRef.current?.focus()
+      }
+    },
+    [wrapperRef, firstValue, secondValue, disabled, isRange, max, step, updateValues],
+  )
+
+  useEffect(() => {
+    window.addEventListener('click', clickOnTrackHandler)
+    return () => {
+      window.removeEventListener('click', clickOnTrackHandler)
+    }
+  }, [clickOnTrackHandler])
+
+  /**
+   * The following callbacks reacts to native HTML events and update the values
+   */
+  const setFirstValueCallback = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const {
+        target: { value },
+      } = e
+      const newValue = parseInt(value, 10)
+      if (newValue > secondValue) {
+        return false
+      }
+      updateValues([newValue, null], e)
+    },
+    [secondValue, updateValues],
+  )
+
+  const setSecondValueCallback = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const {
+        target: { value },
+      } = e
+      const newValue = parseInt(value, 10)
+      if (newValue < firstValue) {
+        return false
+      }
+      updateValues([null, newValue], e)
+    },
+    [firstValue, updateValues],
+  )
+
+  const width: number = ((secondValue - firstValue) / max) * 100
+  const left: number = firstValue / max
+
+  return (
+    <div className={className} data-test={dataTest}>
+      <label id={idRef.current} className={styles.label}>
```

Comment:
Should we use our own `Label`? I understand that we might have two inputs, so it is unclear, which one it should link to. However, the first input is always present, so we could link it to the first one.

Reply:
According to https://www.w3.org/TR/wai-aria-practices-1.2/#slidertwothumb if we have 2 sliders we should reference them through `aria-label` or `aria-labelledby`. This is very special, but we should follow the spec and our current Label component doesn't have this API.

Reply:
Can't we do both? I do not see any contradiction there? Quite the contrary, `label-for` does not prevent us from using `aria-labelledby`, but `label-for` adds easy navigation. Once user clicks on the label, they will be focused on the associated slider thumb. WDYT?

Reply:
Do you mean `htmlFor`? I'm not aware of `label-for`. 

Since there can be only one element with an unique ID, the only solution I see here is to pass this ID to the first slider always. And since the second one doesn't have a visible label, put `aria-label` there. That should follow the spec and fulfill your requirement of using our `Label` thus making clicking on label working.